lib/overcommit/hook/commit_msg/spellcheck.rb

```diff
@@ -0,0 +1,37 @@
+require 'tempfile'
+
+module Overcommit::Hook::CommitMsg
+  # Checks the commit message for potential misspellings
+  class Spellcheck < Base
+    Misspelling = Struct.new(:word, :suggestions)
+
+    MISSPELLING_REGEX = /^[&#]\s(?<word>\w+)(?:.+?:\s(?<suggestions>.*))?/
+
+    def run
+      # Remove comments from commit message file
+      update_commit_message(commit_message)
```

Comment:
I'm realizing now that calling a destructive operation on the commit message file as part of a hook run is dangerous if we were to ever parallelize hook runs.

I realize you didn't add this helper, but I also realize that this helper isn't actually used anywhere (until now).

Ideally, we would create a temporary file with the comment lines removed and run the `hunspell` tool against that.

However, I also acknowledge that `commit-msg` hooks would want to be able to modify the commit message at some point.

For now, since this hook doesn't modify the commit message, let's run `hunspell` against a separate temporary file. We still need to think about the `GerritChangeId` hook, which modifies the commit message file directly.